MM-20255 Update page on Redux actions and remove references to request state (#1100)
mattermost/mattermost-developer-documentation: site/content/contribute/webapp/redux/actions.md

---
title: "Actions"
heading: "Redux Actions"
description: "An explanation of Redux actions and how they're used in Mattermost"
date: 2017-08-20T11:35:32-04:00
weight: 4
---

> Actions in Redux represent an operation either performed by the user or the server which will cause a change to the state of the web app which is stored in the Redux store. It is generally represented as a plain JavaScript object with a constant `type` string and often other data stored in fields such as `data`.

Reviewer: ```suggestion
In Redux, actions represent an operation either performed by the user or the server that cause a change to the state of the web app which is stored in the Redux store. It's generally represented as a plain JavaScript object with a constant `type` string with other data stored in fields such as `data`.
```

```typescript
{
    type: 'SELECT_CHANNEL',
    data: channelId,
}
```

> They are created by functions called action creators. In regular Redux, this function will take some arguments and return an action representing how the store should be changed. Something to note with Mattermost Redux is that we typically refer to the action creators as the "actions" themselves since there's often a single action creator for a given type of action.

Reviewer: ```suggestion
Actions are created by functions called action creators. In regular Redux, this function will take some arguments and return an action representing how the store should be changed. Something to note with Mattermost Redux is that we typically refer to the action creators as the "actions" themselves since there's often a single action creator for a given type of action.
```
